indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3616,8 +3616,26 @@ private void checkIfStreamInactiveAndTurnSupervisorIdle()
       return;
     }
 
-    Map<PartitionIdType, SequenceOffsetType> latestSequencesFromStream = getLatestSequencesFromStream();
     final long nowTime = DateTimes.nowUtc().getMillis();
+    // if it is the first run and there is no lag observed when compared to the offsets from metadata storage stay idle
+    if (!stateManager.isAtLeastOneSuccessfulRun()) {
```

Comment:
I wonder if we should revisit this condition. This will hold true even when there have been multiple runs, none of which have been successful. Would we want to keep checking for idleness then? I guess it's okay. 🤷🏻

Reply:
I guess it is okay to compute this in such cases.

Reply:
can't we just add a check on isFirstRunOnly instead? or is that something else?

Reply:
that just guess that info from the state but doesn't imply as the very first run.